### PR TITLE
Update npd to version v0.8.2

### PR DESCRIPTION
**Pull request by majst01**
update node-problem-detector to v0.8.2 to fix missing hard drive failure detection

/area robustness
/area monitoring
/priority normal

@see https://github.com/kubernetes/node-problem-detector/releases for release notes.

### CODE REVIEW

**Comment by hardikdr** [comment]
/lgtm from changes pov.
Haven't been able to test the happy-path though, can do it later this week, or else please update if already done.

**Comment by majst01** [comment]
@hardikdr we have tested it here by overriding it with the gardenlet image-vector, looks good so far.

**Comment by ialidzhikov** [comment]
Afaik the npd chart was copied from helm/charts. So there are no changes in the chart itself between this version range?

**Comment by majst01** [comment]
@ialidzhikov thanks for the hint, in fact there is one more volume `/dev/kmsg` and NODE_NAME env, added these from upstream

File: charts/shoot-core/components/charts/node-problem-detector/templates/daemonset.yaml [hunk 1]
```diff
@@ -98,6 +101,9 @@ spec:
           hostPath:
             path: /etc/localtime
             type: "FileOrCreate"
+        - name: kmsg
```
Comment (timebertt) [inline charts/shoot-core/components/charts/node-problem-detector/templates/daemonset.yaml:104]: I think, the PSP has to be adapted to this change, right?
https://github.com/gardener/gardener/blob/78ef5841ad5a0d9ba405a5ccb66b0e9869a9f390/charts/shoot-core/components/charts/node-problem-detector/templates/psp.yaml#L25-L27

Comment (majst01) [reply charts/shoot-core/components/charts/node-problem-detector/templates/daemonset.yaml:104]: huh, thanks for pointing out

File: charts/shoot-core/components/charts/node-problem-detector/values.yaml [hunk 2]
```diff
@@ -86,6 +86,10 @@ affinity: {}
 nodeSelector: {}
 
 env:
+  - name: NODE_NAME
+    valueFrom:
+      fieldRef:
+        fieldPath: spec.nodeName
```
Comment (timebertt) [inline charts/shoot-core/components/charts/node-problem-detector/values.yaml]: We already have the env var here:
https://github.com/gardener/gardener/blob/2e993388559b10582f1fc00623cb371b5262df65/charts/shoot-core/components/charts/node-problem-detector/templates/daemonset.yaml#L49-L53
Why is this added a second time in the values?

This way, the env var will be added twice 😄 
```yaml
        env:
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
```

**Review by timebertt — CHANGES_REQUESTED** [review]
Thanks for the PR, only one nit.
Tested and validated successfully, that NPD is running fine on an unprivileged cluster. :)

Can you please squash your commits then to get the PR ready for merging?

**Comment by timebertt** [comment]
Actually, it would be really nice to have some kind of integration test for this kind of stuff like the NPD.
The test could provoke issues on the Nodes, that the NPD should detect and care controller should pick up and reflect in the Shoot status.
Ref #1564

**Comment by timuthy** [comment]
> Actually, it would be really nice to have some kind of integration test for this kind of stuff like the NPD.
The test could provoke issues on the Nodes, that the NPD should detect and care controller should pick up and reflect in the Shoot status.

+1 but let's do it in a separate PR to not block this one from getting merged.

**Comment by timebertt** [comment]
> > Actually, it would be really nice to have some kind of integration test for this kind of stuff like the NPD.
> > The test could provoke issues on the Nodes, that the NPD should detect and care controller should pick up and reflect in the Shoot status.
> 
> +1 but let's do it in a separate PR to not block this one from getting merged.

Sure, didn't want to imply that we should do it in this PR. Just a side-note :)

**Comment by majst01** [comment]
I am to stupid to squash, even googling makes me sad :-(

**Review by timebertt — DISMISSED** [review]
Thanks!
/lgtm now

**Comment by timebertt** [comment]
> I am to stupid to squash, even googling makes me sad :-(

Don't be sad :)
The easy option would be:
1. `git reset --soft master`
2. `git commit`
3. `git push -f fork`

Or you can go for:
1. `git rebase -i master`
2. Edit the opened file use `pick` for the first commit (the original one) and replace all `pick`s of consecutive commits with `squash`/`s`.
3. Do a final edit of the commit message
4. `git push -f fork`

Does this help?

**Comment by majst01** [comment]
@tim-ebert Thanks! this was the best explanation so far.

**Review by timebertt — APPROVED** [review]
Nice! :)
/lgtm
/ok-to-test

**Comment by majst01** [comment]
Thanks @tim-ebert
